core/src/main/scala/pink/cozydev/lucille/Query.scala

```diff
@@ -20,6 +20,19 @@ import cats.data.NonEmptyList
 
 sealed trait Query extends Product with Serializable
 
+final case class MultiQuery(qs: NonEmptyList[Query]) extends Query {
+  def mapLast(f: Query => Query): MultiQuery =
+    if (qs.size == 1) MultiQuery(NonEmptyList.one(f(qs.head)))
+    else {
+      val newT = qs.tail.init :+ f(qs.last)
```

Comment:
I always forget about `init`, nice

Reply:
I don't really understand the naming, but yeah, it's handy.

Sucks that changing the last element is so expensive though.